text/0000-raw-txn-header.md

```diff
@@ -0,0 +1,96 @@
+- Feature Name: send_serialize_txn
+- Start Date: 2018-08-03
+- RFC PR:
+- Sawtooth Issue:
+
+# Summary
+[summary]: #summary
+
+Requesting to add API to the sdk - txn.get_serialized_header(), that will return
+the transaction header bytes.
+
+# Motivation 
+[motivation]: #motivation
+
+For security reasons transaction processor would like to verify the incoming
+transaction header, and signature.  For example, in the private ledger TP (C++)
+that runs inside SGX enclave, we can't trust the signature verification done by
+sawtooth since it happens outside of SGX.  In order to verify the transaction
+signature, TP needs to get the serialized transaction header and not just all
+the fields.
```

Comment:
This does not explain why it is important to be able to verify the signature within the TEE.

It seems like this could have been solved entirely within the payload content.

Reply:
payload is verified with its hash that is in the signed txn header, TP needs a way to detect that attacker didn't replace the txn payload from different txn, only way to know that is if we check signature of header that includes the payload hash